src/tree/updater_quantile_hist.h

```diff
@@ -158,9 +158,12 @@ class QuantileHistMaker: public TreeUpdater {
                           const GHistIndexMatrix& gmat,
                           const GHistIndexBlockMatrix& gmatb,
                           GHistRow hist) {
+//      std::cout  << "\n_____QuantileHistMaker::Builder::BuildHist\n";
       if (param_.enable_feature_grouping > 0) {
+//        std::cout << "\n ??? WTF ??? \n";
```

Comment:
:D

Reply:
Finally this line became clear for me ;)